Automattic/newspack-newsletters
# Add tags support to Constant Contact

### All Submissions:






### Changes proposed in this Pull Request:



Adds support to Tags for the Constant Contact provider. This will enable the new Lists feature for this provider.

### How to test the changes in this Pull Request:

1. Set Constant Contact as the provider with proper credentials
2. Go to Newsletter > Lists
3. Create a new list
4. Confirm that a correspondent tag is created in the provider
5. Go to Newsletter > Settings and enable the list in the Subsctiption Lists section
6. Add a registration or Newsletter subscription block to your site with that list as an option
7. As a visitor, register to a newsletter selecting that list
8. confirm the contact is created with the correct tag in the provider
9. confirm also that a Segment was created with users tagged with this tag
10. (Note: Constant Contact is not supported in the "My account" page for managing subscriptions, this is unrelated with this PR)
11. As a failsafe, make sure the same steps above are still working with other providers

Also note that the ability to select this segment in the editor when sending newsletters will be added in another PR (UPDATE: It was added in https://github.com/Automattic/newspack-newsletters/pull/1055)

### Other information:
